docs/projects/selection-sort/_posts/2019-01-10-python.md

```diff
@@ -0,0 +1,176 @@
+---
+title: Selection Sort in Python
+layout: default
+last-modified: 2019-01-10
+featured-image: selection-sort-in-python-featured-image.JPEG
+tags: [python, selection-sort]
+authors:
+  - hmajid2301
+---
+  
+In this article, we'll be tackling Selection Sort in Python.
+
+## How to Implement the Solution
+
+At this point, let's dig into the code a bit. The following sections break
+down the Selection Sort in Python functionality.
+
+### Solution
+
+```python
+#!/usr/bin/env python
+import sys
+
+
+def selection_sort(xs, sorted_xs=None):
+    sorted_xs = sorted_xs or []
+    if len(xs) <= 0:
+        return sorted_xs
+    x = min(xs)
+    sorted_xs.append(x)
+    xs.remove(x)
+    return selection_sort(xs, sorted_xs)
+
+
+def input_list(list_str):
+    return [int(x.strip(" "), 10) for x in list_str.split(',')]
+
+
+def exit_with_error():
+    print('Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”')
+    sys.exit(1)
+
+
+def main(args):
+    try:
+        xs = input_list(args[0])
+        if len(xs) <= 1:
+            exit_with_error()
+        print(selection_sort(xs))
+    except (IndexError,ValueError):
+        exit_with_error()
+
+
+if __name__ == "__main__":
+    main(sys.argv[1:])
+```
+
+### main()
```

Comment:
I prefer headings to be more general and less code-specific. Instead of "main()", it may look cleaner with something like "The Main Function".

Of course, that's my personal preference, so take it with a grain of salt.

Reply:
Makes sense, I agree with you.

Reply:
Visual Studio Code wraps my code so I never notice the line length lol. Will fix it :)